### PR TITLE
doc: revise Processing docstrings for formatting and class links

### PR DESCRIPTION
**Pull request by laurenyu**
*Issue #, if available:*

*Description of changes:*

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have used the regional endpoint when creating S3 and/or STS clients (if appropriate)
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/processing.py [hunk 1]
```diff
@@ -48,40 +48,43 @@ def __init__(
...
-            tags ([dict]): List of tags to be passed to the processing job.
-            network_config (sagemaker.network.NetworkConfig): A NetworkConfig
+                processing image name and current timestamp.
+            sagemaker_session (:class:`~sagemaker.session.Session`):
+                Session object which manages interactions with Amazon SageMaker and
+                any other AWS services needed. If not specified, the processor creates
+                one using the default AWS configuration chain.
+            env (dict[str, str]): Environment variables to be passed to
+                the processing jobs (default: None).
+            tags (list[dict]): List of tags to be passed to the processing job.
```
Comment (knakad) [inline src/sagemaker/processing.py]:
```suggestion
            tags (list[dict]): List of tags to be passed to the processing job (default: None).
```

File: src/sagemaker/processing.py [hunk 2]
```diff
@@ -179,15 +185,18 @@ def _generate_current_job_name(self, job_name=None):
         return name_from_base(base_name)
 
     def _normalize_inputs(self, inputs=None):
-        """Ensure that all the ProcessingInput objects have names and S3 uris.
+        """Ensure that all the ``ProcessingInput`` objects have names and S3 URIs.
 
         Args:
-            inputs ([sagemaker.processing.ProcessingInput]): A list of ProcessingInput
+            inputs (list[sagemaker.processing.ProcessingInput]): A list of ``ProcessingInput``
                 objects to be normalized.
```
Comment (knakad) [inline src/sagemaker/processing.py]:
```suggestion
                objects to be normalized (default: None).
```

File: src/sagemaker/processing.py [hunk 3]
```diff
@@ -220,17 +229,20 @@ def _normalize_inputs(self, inputs=None):
...
-        names and S3 uris.
+        """Ensure that all the outputs are ``ProcessingOutput`` objects with
+        names and S3 URIs.
 
         Args:
-            outputs ([sagemaker.processing.ProcessingOutput]): A list
+            outputs (list[sagemaker.processing.ProcessingOutput]): A list
                 of outputs to be normalized. Can be either strings or
-                ProcessingOutput objects.
+                ``ProcessingOutput`` objects.
```
Comment (knakad) [inline src/sagemaker/processing.py]:
```suggestion
                ``ProcessingOutput`` objects (default: None).
```

File: src/sagemaker/processing.py [hunk 4]
```diff
@@ -277,41 +289,43 @@ def __init__(
...
-            tags ([dict]): List of tags to be passed to the processing job.
-            network_config (sagemaker.network.NetworkConfig): A NetworkConfig
+                processing image name and current timestamp.
+            sagemaker_session (:class:`~sagemaker.session.Session`):
+                Session object which manages interactions with Amazon SageMaker and
+                any other AWS services needed. If not specified, the processor creates
+                one using the default AWS configuration chain.
+            env (dict[str, str]): Environment variables to be passed to
+                the processing jobs (default: None).
+            tags (list[dict]): List of tags to be passed to the processing job.
```
Comment (knakad) [inline src/sagemaker/processing.py]:
```suggestion
            tags (list[dict]): List of tags to be passed to the processing job (default: None).
```

File: src/sagemaker/processing.py [hunk 5]
```diff
@@ -346,25 +360,26 @@ def run(
...
-            code (str): This can be an S3 uri or a local path to either
-                a directory or a file with the user's script to run.
-            inputs ([sagemaker.processing.ProcessingInput]): Input files for the processing
-                job. These must be provided as ProcessingInput objects.
-            outputs ([str or sagemaker.processing.ProcessingOutput]): Outputs for the processing
-                job. These can be specified as either a path string or a ProcessingOutput
-                object.
-            arguments ([str]): A list of string arguments to be passed to a
-                processing job.
+            code (str): This can be an S3 URI or a local path to either
```
Comment (knakad) [inline src/sagemaker/processing.py]: @ajaykarpur I believe this must be a file (not a directory), now, correct?

Comment (ajaykarpur) [reply src/sagemaker/processing.py]: That's correct (take a look at `_handle_user_code_url`).

**Pull request merged**